modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -527,6 +531,463 @@ public void testAddCustomObjectField() throws Exception {
+
+		objectField = _addCustomObjectField(
+			new FormulaObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					new ObjectFieldSettingBuilder(
+					).name(
+						"script"
+					).value(
+						"weight + 10"
+					).build(),
+					new ObjectFieldSettingBuilder(
+					).name(
+						"output"
+					).value(
+						ObjectFieldConstants.BUSINESS_TYPE_DECIMAL
+					).build())
+			).readOnly(
+				ObjectFieldConstants.READ_ONLY_FALSE
+			).build());
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		objectField = _addCustomObjectField(
+			new FormulaObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					new ObjectFieldSettingBuilder(
+					).name(
+						"script"
+					).value(
+						"weight + 10"
+					).build(),
+					new ObjectFieldSettingBuilder(
+					).name(
+						"output"
+					).value(
+						ObjectFieldConstants.BUSINESS_TYPE_DECIMAL
+					).build())
+			).readOnly(
+				RandomTestUtil.randomString()
+			).build());
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		objectField = _addCustomObjectField(
+			new FormulaObjectFieldBuilder(
+			).labelMap(
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString())
+			).name(
+				"a" + RandomTestUtil.randomString()
+			).objectDefinitionId(
+				objectDefinition1.getObjectDefinitionId()
+			).objectFieldSettings(
+				Arrays.asList(
+					new ObjectFieldSettingBuilder(
+					).name(
+						"script"
+					).value(
+						"weight + 10"
+					).build(),
+					new ObjectFieldSettingBuilder(
+					).name(
+						"output"
+					).value(
+						ObjectFieldConstants.BUSINESS_TYPE_DECIMAL
+					).build())
+			).readOnly(
+				ObjectFieldConstants.READ_ONLY_CONDITIONAL
+			).readOnlyConditionExpression(
+				"isEmpty(able)"
+			).build());
+
+		Assert.assertEquals(
+			objectField.getReadOnly(), ObjectFieldConstants.READ_ONLY_TRUE);
+		Assert.assertEquals(
+			objectField.getReadOnlyConditionExpression(), StringPool.BLANK);
+
+		ObjectDefinition objectDefinition2 =
+			ObjectDefinitionTestUtil.addObjectDefinition(
+				false, _objectDefinitionLocalService,
+				Arrays.asList(
+					new IntegerObjectFieldBuilder(
+					).labelMap(
+						LocalizedMapUtil.getLocalizedMap(
+							RandomTestUtil.randomString())
+					).name(
+						"integerObjectFieldName"
```

Comment:
You can create a variable to store this value, you are using it six times.
Do you agree?